### PR TITLE
Feature/resources strategy

### PR DESCRIPTION
**Pull request by justinjj24**
Added error strategy and suitable resources for each process..
- disk space
- error strategy exit codes and action (ignore/finish)
- maxretry to 2
- apply matching cpus to process
- added `--NumThread ${vf_threads}` to verifybamid2 process to double the `tasks.cpus`

### CODE REVIEW

File: modules/verifybamid2/main.nf [hunk 1]
```diff
@@ -16,11 +15,13 @@ process verifybamid2 {
 
     script:
     def reference = ref_fasta ? /--Reference "${ref_fasta}"/ : ''
+
+    vf_threads = (int) Math.ceil(task.cpus*2)
```
Comment (mhebrard) [inline modules/verifybamid2/main.nf:19]: This is a lot of code for just *2 a variable... I dont think it is needed

File: modules/verifybamid2/main.nf [hunk 2]
```diff
@@ -16,11 +15,13 @@ process verifybamid2 {
...
     """
    # run verifybamid2
    # get the percentage of cross-individual contamination rate 
     
-    verifybamid2 --NumPC 4 --SVDPrefix ${params.vbi2_svdprefix} ${reference} --BamFile ${bam} --Output ${sample}
+    verifybamid2 --NumPC 4 --NumThread ${vf_threads} --SVDPrefix ${params.vbi2_svdprefix} ${reference} --BamFile ${bam} --Output ${sample}
```
Comment (mhebrard) [inline modules/verifybamid2/main.nf:24]: This should suffice:
```suggestion
    verifybamid2 --NumPC 4 --NumThread ${task.cpus*2} --SVDPrefix ${params.vbi2_svdprefix} ${reference} --BamFile ${bam} --Output ${sample}
```

**Review by mhebrard — CHANGES_REQUESTED** [review]

File: modules/verifybamid2/main.nf (hunk 2 above)
Comment (justinjj24) [reply modules/verifybamid2/main.nf:24]: strange! when I test last week it `${task.cpus*2}` wasn't worked then I add the `${vf_threads}`

File: modules/verifybamid2/main.nf (hunk 1 above)
Comment (justinjj24) [reply modules/verifybamid2/main.nf]: suggestion updated and tested

**Review by mhebrard — APPROVED** [review]

**Pull request merged**